### PR TITLE
Document get.pulumi.com to download PRs

### PR DESCRIPTION
**Pull request by Frassle**
As suggested by @tgummerer at https://github.com/pulumi/pulumi/pull/17015/files#r1722817544.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -147,6 +147,14 @@ Here's some examples of what we're trying to avoid:
 - Adds a feature
 - Feature now does something
 
+### Downloading Pulumi from contributed pull requests
+
+Artifacts built during pull request workflows can be downloaded by running the following command:
+
+```sh
+curl -fsSL https://get.pulumi.com | sh -s -- --version pr#NUMBER
```
Comment (justinvp) [inline CONTRIBUTING.md:155]: I was initially confused whether the `#` is necessary… wondering if showing it this way would be more clear?

```suggestion
curl -fsSL https://get.pulumi.com | sh -s -- --version pr#<number>
```

**Review by justinvp — APPROVED** [review]

File: CONTRIBUTING.md (hunk 1 above)
Comment (Frassle) [reply CONTRIBUTING.md:155]: oh yeh that's better

**Review by tgummerer — APPROVED** [review]
Thanks for documenting this.  I've disabled auto-merge to consider the comment inline, but feel free to ignore it if you don't think it's worth it.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -147,6 +147,14 @@ Here's some examples of what we're trying to avoid:
 - Adds a feature
 - Feature now does something
 
+### Downloading Pulumi from contributed pull requests
+
+Artifacts built during pull request workflows can be downloaded by running the following command:
```
[the remark this answers is not in the document]
Comment (Frassle) [reply CONTRIBUTING.md:152]: Yeh 1 day seems a bit short, especially given the timezone differences we have across our team and users. Maybe 5 day retention (i.e. a work week) and add the comment here that they expire after 5 days?

[the remark this answers is not in the document]
Comment (tgummerer) [reply CONTRIBUTING.md:152]: I think I'd just make it 7 as that's somewhat easier to explain, but either works for me!

[the remark this answers is not in the document]
Comment (Frassle) [reply CONTRIBUTING.md:152]: Sure

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -147,6 +147,14 @@ Here's some examples of what we're trying to avoid:
 - Adds a feature
 - Feature now does something
 
+### Downloading Pulumi from contributed pull requests
+
+Artifacts built during pull request workflows can be downloaded by running the following command (note that the artifacts expire 7 days after CI has been run)::
```
Comment (tgummerer) [inline CONTRIBUTING.md:152]:
```suggestion
Artifacts built during pull request workflows can be downloaded by running the following command (note that the artifacts expire 7 days after CI has been run):
```

**Review by tgummerer — APPROVED** [review]
Looks great, thanks!

**Pull request merged**